src/hardware/disk_noise.cpp

```diff
@@ -0,0 +1,479 @@
+/*
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *
+ *  Copyright (C) 2025-2025  The DOSBox Staging Team
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+#include "disk_noise.h"
+
+#include <algorithm>
+#include <array>
+#include <cmath>
+#include <cstring>
+#include <fstream>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "../libs/decoders/dr_flac.h"
+#include "channel_names.h"
+#include "checks.h"
+#include "mixer.h"
+#include "setup.h"
+#include "timer.h"
+
+CHECK_NARROWING();
+
+// Static member definitions
+std::shared_ptr<MixerChannel> DiskNoiseDevice::mix_channel_ = nullptr;
+std::vector<DiskNoiseDevice*> DiskNoiseDevice::active_devices_;
+std::mutex DiskNoiseDevice::device_mutex_;
```

Comment:
We try to keep such static stuff to the bare minimum in new code. It would be better to introduce a new holder class for these things, e.g. `DiskNoises`. You don't need separate cpp/h files, they can go here.

Then one single `unique_ptr<DiskNoises>` in the global scope, that's cleaner and more encapsulated.